tico/serialize/circle_graph.py

```diff
@@ -241,6 +250,15 @@ def add_tensor_from_scratch(
         if source_node is not None:
             self.name_to_node[tensor.name] = source_node
         tensor.shape = shape
+
+        # Handle dynamic shape
+        if any(isinstance(s, torch.SymInt) for s in tensor.shape):
```

Comment:
The type hints and comments of this function have to be updated as well.

```python
        shape: List[int],
```

```
        shape : List[int]
            The shape of the tensor.
```

Reply:
Let me revisit there in the next PR!